Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -0,0 +1,469 @@
+  use K, hK
+  intro j
+  obtain ⟨N, hN⟩ := cover (ε/(2^(j+1))) (hεj_pos j) (1/(j+1)) (Nat.one_div_pos_of_nat)
+  use N
+  rw [measure_compl (by measurability) (measure_ne_top μ _)]
+  exact tsub_le_iff_tsub_le.mp hN
+
+/-- A measure `μ` is separable if there is a separable set `S` such that `μ S = μ Set.univ`. -/
+ def IsSeparable [TopologicalSpace α] (μ : Measure α) : Prop :=
+   ∃ S : Set α, TopologicalSpace.IsSeparable S ∧ μ S = μ Set.univ
+
+namespace IsSeparable
+
+lemma of_separableSpace [TopologicalSpace α] [TopologicalSpace.SeparableSpace α] :
+    IsSeparable μ := ⟨Set.univ, TopologicalSpace.isSeparable_univ_iff.mpr ‹_›, rfl⟩
+
+end IsSeparable
+
+/-- A measure `μ` is pre-tight if for all `0 < ε`, there exists `K` totally bounded such that
+  `μ Kᶜ ≤ ε`. -/
+def IsPretight [UniformSpace α] (μ : Measure α) : Prop :=
+  ∀ ε : ℝ≥0∞, 0 < ε → ∃ K : Set α, TotallyBounded K ∧ μ Kᶜ ≤ ε
+
+namespace IsPretight
+
+lemma of_totallyBounded [UniformSpace α] (h : TotallyBounded (Set.univ : Set α)) :
+    IsPretight μ := by
+  intro ε hε
+  use Set.univ, h
+  rw [Set.compl_univ, measure_empty]
+  exact hε.le
+
+lemma exists_totallyBounded_nat [UniformSpace α] (h : IsPretight μ) :
+    ∀ n : ℕ, ∃ K : Set α, TotallyBounded K ∧ μ Kᶜ ≤ 1/n := by
+  intro n
+  apply h
+  simp
+
+/-- It suffices to check totally boundedness along countably many `ε`. -/
+lemma of_totallyBounded_nat [UniformSpace α]
+    (h : ∀ n : ℕ, ∃ K : Set α, TotallyBounded K ∧ μ Kᶜ ≤ 1/n) : IsPretight μ := by
+  intro ε hε
+  obtain ⟨n, hn⟩ := ENNReal.exists_inv_nat_lt hε.ne'
+  obtain ⟨K, hK, hKe⟩ := h n
+  refine ⟨K, hK, ?_⟩
+  apply le_trans hKe (le_trans ?_ hn.le)
+  rw [one_div, ENNReal.inv_le_inv]
+
+lemma totallyBounded_nat_iff [UniformSpace α] :
+    IsPretight μ ↔ ∀ n : ℕ, ∃ K : Set α, TotallyBounded K ∧ μ Kᶜ ≤ 1/n :=
+  ⟨exists_totallyBounded_nat, of_totallyBounded_nat⟩
+
+/-- If a measure `μ` is pretight, we can cover `μ`-almost all of the space by a countable sequence
+of totally bounded sets. -/
+lemma exists_countable_totallyBounded_union [UniformSpace α] (h : IsPretight μ):
+    ∃ K : ℕ → Set α, (∀ n, TotallyBounded (K n)) ∧ μ (⋃ n, K n) = μ Set.univ := by
+  choose! K hKa hKb using h.exists_totallyBounded_nat
+  use K, hKa
+  rw [← Set.iUnion_accumulate, measure_congr]
+  rw [ae_eq_univ, Set.compl_iUnion, ← le_zero_iff]
+  refine le_of_forall_lt' (fun ε hε ↦ ?_)
+  obtain ⟨n, hn⟩ := ENNReal.exists_inv_nat_lt hε.ne.symm
+  rw [← one_div] at hn
+  have : μ ((Set.Accumulate K n)ᶜ) ≤ μ ((K n)ᶜ) := by
+    apply measure_mono
+    rw [Set.compl_subset_compl]
+    exact Set.subset_accumulate
+  exact lt_of_le_of_lt (measure_mono <| Set.iInter_subset _ n)
+    (lt_of_le_of_lt this (lt_of_le_of_lt (hKb n) hn))
+
+/-- Every pretight measure on a countably generated uniform space is separable. -/
+lemma to_isSeparable_on_countably_generated_uniformSpace [UniformSpace α]
+    [i : IsCountablyGenerated (𝓤 α)] (h : IsPretight μ) : IsSeparable μ := by
```

Comment:
Why are you naming the `i` instance? If you don't need a name for instances, don't provide one. Same thing below.